File: A. Easy/0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortedSquares(int* nums, int numsSize, int* returnSize) {
    int* numsSquares = (int*)malloc(numsSize * sizeof(int));
    if (numsSquares == NULL) {
        *returnSize = 0;
        return NULL;
    }

    int left = 0, right = numsSize - 1;
    int i = numsSize - 1;

    while (left <= right) {
        int leftSquare = nums[left] * nums[left];
        int rightSquare = nums[right] * nums[right];

        if (leftSquare > rightSquare) {
            numsSquares[i] = leftSquare;
            left++;
        }
        else {
            numsSquares[i] = rightSquare;
            right--;
        }
        i--;
    }
    *returnSize = numsSize;
    return numsSquares;
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        numsSquares = list()
        while (left <= right):
            leftSquare = nums[left] ** 2
            rightSquare = nums[right] ** 2
            if (leftSquare > rightSquare):
                numsSquares.append(leftSquare)
                left += 1
            else:
                numsSquares.append(rightSquare)
                right -= 1
        return numsSquares[::-1]
        